test/infrastructure/docker/api/v1beta1/dockermachinetemplate_webhook.go

```diff
@@ -17,48 +17,50 @@ limitations under the License.
 package v1beta1
 
 import (
+	"context"
 	"fmt"
 	"reflect"
 
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
 
 	"sigs.k8s.io/cluster-api/util/webhooks"
 )
 
 const dockerMachineTemplateImmutableMsg = "DockerMachineTemplate spec.template.spec field is immutable. Please create a new resource instead."
 
 func (m *DockerMachineTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
-	// Register the topology aware validator webhook.
-	mgr.GetWebhookServer().Register(
-		"/validate-infrastructure-cluster-x-k8s-io-v1beta1-dockermachinetemplate",
-		webhooks.TopologyAwareValidatingWebhookFor(m),
-	)
-	// Register the defaulter webhook.
 	return ctrl.NewWebhookManagedBy(mgr).
+		WithValidator(m).
 		For(m).
 		Complete()
 }
 
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-dockermachinetemplate,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=dockermachinetemplates,versions=v1beta1,name=validation.dockermachinetemplate.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 
-var _ webhooks.TopologyAwareValidator = &DockerMachineTemplate{}
+var _ webhook.CustomValidator = &DockerMachineTemplate{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
-func (m *DockerMachineTemplate) ValidateCreate() error {
+func (*DockerMachineTemplate) ValidateCreate(ctx context.Context, _ runtime.Object) error {
```

Comment:
Can we intentionally make this a separate struct? (cf. my PR on your fork) In the other cases we use the CustomValidator we use a separate struct for it because we have to, but I think here it would be good to avoid confusion, for consistency with other CustomValidator webhooks and to avoid additional methods on the API type